smartconfig/file.py

```diff
@@ -1,13 +1,19 @@
 from os import PathLike
-from typing import Mapping, MutableMapping, Union
+from typing import Mapping, MutableMapping, Set, Union
 
 import yaml
 
-from smartconfig import _registry
+from smartconfig import ConfigurationError, _registry
+from smartconfig._registry import used_paths
 from smartconfig.constructors import _ref_constructor
 
 
-def _update_mapping(source: Mapping, dest: MutableMapping, path: str = "") -> MutableMapping:
+def _update_mapping(
+        source: Mapping,
+        dest: MutableMapping,
+        path: str = "",
+        node_paths: Set[str] = set()  # noqa: B006, this is safe since we don't mutate it
```

Comment:
The default value doesn't seem to be used anywhere. Why does it even need to be a parameter if you're only using `used_paths`?

Reply:
I initially made it a default argument because of the two recursive calls, that's just my brain being smooth. I like the idea of having a default argument here though, this is a more general function that can be used elsewhere.

Reply:
I see your point, but I would've rather not have it added, especially considering it's a private utility function only used once. However, since you already did, I guess it can stay.